### PR TITLE
MINOR: SRE-3220: ignore lifecycle changes for backup location

### PR DESCRIPTION
**Pull request by ApekshaKhilari**
https://cloud.google.com/sql/docs/mysql/backup-recovery/backups#default-backup-location

### CODE REVIEW

**Comment by tamr-github-integrator** [comment]
# [tfsec] Results
## Failed: 14 issue(s)
| # | ID | Severity | Title | Location | Description |
|---|----|----------|-------|----------|-------------|
| 1 | `google-sql-enable-pg-temp-file-logging` | *MEDIUM* | _Temporary file logging should be enabled for all temporary files._ | `main.tf:8-65` | Database instance has temporary file logging disabled. |
| 2 | `google-sql-encrypt-in-transit-data` | *HIGH* | _SSL connections to a SQL database instance should be enforced._ | `main.tf:8-65` | Database instance does not require TLS for all connections. |
| 3 | `google-sql-no-public-access` | *HIGH* | _Ensure that Cloud SQL Database Instances are not publicly exposed_ | `main.tf:8-65` | Database instance is granted a public internet address. |
| 4 | `google-sql-pg-log-checkpoints` | *MEDIUM* | _Ensure that logging of checkpoints is enabled._ | `main.tf:8-65` | Database instance is not configured to log checkpoints. |
| 5 | `google-sql-pg-log-connections` | *MEDIUM* | _Ensure that logging of connections is enabled._ | `main.tf:8-65` | Database instance is not configured to log connections. |
| 6 | `google-sql-pg-log-disconnections` | *MEDIUM* | _Ensure that logging of disconnections is enabled._ | `main.tf:8-65` | Database instance is not configured to log disconnections. |
| 7 | `google-sql-pg-log-lock-waits` | *MEDIUM* | _Ensure that logging of lock waits is enabled._ | `main.tf:8-65` | Database instance is not configured to log lock waits. |
| 8 | `google-sql-enable-pg-temp-file-logging` | *MEDIUM* | _Temporary file logging should be enabled for all temporary files._ | `main.tf:8-65` | Database instance has temporary file logging disabled. |
| 9 | `google-sql-encrypt-in-transit-data` | *HIGH* | _SSL connections to a SQL database instance should be enforced._ | `main.tf:8-65` | Database instance does not require TLS for all connections. |
| 10 | `google-sql-no-public-access` | *HIGH* | _Ensure that Cloud SQL Database Instances are not publicly exposed_ | `main.tf:8-65` | Database instance is granted a public internet address. |
| 11 | `google-sql-pg-log-checkpoints` | *MEDIUM* | _Ensure that logging of checkpoints is enabled._ | `main.tf:8-65` | Database instance is not configured to log checkpoints. |
| 12 | `google-sql-pg-log-connections` | *MEDIUM* | _Ensure that logging of connections is enabled._ | `main.tf:8-65` | Database instance is not configured to log connections. |
| 13 | `google-sql-pg-log-disconnections` | *MEDIUM* | _Ensure that logging of disconnections is enabled._ | `main.tf:8-65` | Database instance is not configured to log disconnections. |
| 14 | `google-sql-pg-log-lock-waits` | *MEDIUM* | _Ensure that logging of lock waits is enabled._ | `main.tf:8-65` | Database instance is not configured to log lock waits. |

File: variables.tf [hunk 1]
```diff
@@ -62,6 +62,12 @@ variable "backup_enabled" {
   default     = true
 }
 
+variable "backup_location" {
+  type        = string
+  description = "The region where the backup will be stored. If not specified, backups will be stored in the multiregion geographically nearest to the instance"
+  default     = ""
```
Comment (laferrieren) [inline variables.tf:68]: You should either set this a specific location or remove this so a user knows that it is required

**Review by laferrieren — COMMENTED** [review]
please make sure to run make terraform/docs

File: variables.tf (hunk 1 above)
Comment (ApekshaKhilari) [reply variables.tf:68]: I wanted to keep it backward compatible for all other places where its used (& this module is used at several places). If its left empty, it'll use default location - that is create backups in a multiregion that is geographically closest to the location where the clousql instance is created. Just wanted to have a functionality to set custom location as well.

**Comment by tamr-github-integrator** [comment]
# [tfsec] Results
## Failed: 14 issue(s)
| # | ID | Severity | Title | Location | Description |
|---|----|----------|-------|----------|-------------|
| 1 | `google-sql-enable-pg-temp-file-logging` | *MEDIUM* | _Temporary file logging should be enabled for all temporary files._ | `main.tf:8-70` | Database instance has temporary file logging disabled. |
| 2 | `google-sql-encrypt-in-transit-data` | *HIGH* | _SSL connections to a SQL database instance should be enforced._ | `main.tf:8-70` | Database instance does not require TLS for all connections. |
| 3 | `google-sql-no-public-access` | *HIGH* | _Ensure that Cloud SQL Database Instances are not publicly exposed_ | `main.tf:8-70` | Database instance is granted a public internet address. |
| 4 | `google-sql-pg-log-checkpoints` | *MEDIUM* | _Ensure that logging of checkpoints is enabled._ | `main.tf:8-70` | Database instance is not configured to log checkpoints. |
| 5 | `google-sql-pg-log-connections` | *MEDIUM* | _Ensure that logging of connections is enabled._ | `main.tf:8-70` | Database instance is not configured to log connections. |
| 6 | `google-sql-pg-log-disconnections` | *MEDIUM* | _Ensure that logging of disconnections is enabled._ | `main.tf:8-70` | Database instance is not configured to log disconnections. |
| 7 | `google-sql-pg-log-lock-waits` | *MEDIUM* | _Ensure that logging of lock waits is enabled._ | `main.tf:8-70` | Database instance is not configured to log lock waits. |
| 8 | `google-sql-enable-pg-temp-file-logging` | *MEDIUM* | _Temporary file logging should be enabled for all temporary files._ | `main.tf:8-70` | Database instance has temporary file logging disabled. |
| 9 | `google-sql-encrypt-in-transit-data` | *HIGH* | _SSL connections to a SQL database instance should be enforced._ | `main.tf:8-70` | Database instance does not require TLS for all connections. |
| 10 | `google-sql-no-public-access` | *HIGH* | _Ensure that Cloud SQL Database Instances are not publicly exposed_ | `main.tf:8-70` | Database instance is granted a public internet address. |
| 11 | `google-sql-pg-log-checkpoints` | *MEDIUM* | _Ensure that logging of checkpoints is enabled._ | `main.tf:8-70` | Database instance is not configured to log checkpoints. |
| 12 | `google-sql-pg-log-connections` | *MEDIUM* | _Ensure that logging of connections is enabled._ | `main.tf:8-70` | Database instance is not configured to log connections. |
| 13 | `google-sql-pg-log-disconnections` | *MEDIUM* | _Ensure that logging of disconnections is enabled._ | `main.tf:8-70` | Database instance is not configured to log disconnections. |
| 14 | `google-sql-pg-log-lock-waits` | *MEDIUM* | _Ensure that logging of lock waits is enabled._ | `main.tf:8-70` | Database instance is not configured to log lock waits. |

**Review by tparikh — APPROVED** [review]

**Pull request merged**